buffer_test.go

```diff
@@ -95,3 +96,60 @@ func TestBufferSafety(t *testing.T) {
 	t.Log("post gc 2")
 	verify()
 }
+
+type ByteCounter struct {
+	BytesWritten int64
+}
+
+func (b *ByteCounter) Write(x []byte) (int, error) {
+	b.BytesWritten += int64(len(x))
+	return len(x), nil
+}
+
+func TestWriteTo(t *testing.T) {
+	context, err := NewContext(nil)
+	require.NoError(t, err)
+	buffer, err := NewBuffer(context)
+	require.NoError(t, err)
+
+	testSizes := [5]int{0, 16, 256, 65536, 268435456}
```

Comment:
Nit: I like to use separators to make large numbers easier to parse visually, but that's just personal taste. Feel free to ignore.
```suggestion
	testSizes := [5]int{0, 16, 256, 65536, 268_435_456}
```

Reply:
Updated to `256 << 20` to more clearly state where this number comes from.